CMakePresets.json

```diff
@@ -0,0 +1,63 @@
+{
+    "version": 5,
+    "configurePresets": [
+        {
+            "name": "common",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_COMPILER_LAUNCHER": "sccache"
+            }
```

Comment:
Won't this break if the user doesn't have sccache installed?

Reply:
Right, and if you are missing ninja. I could add more presets for the different options like other projects do...

Reply:
Added different presets for different combinations.